cli/lsp/documents.rs

```diff
@@ -1057,61 +1100,99 @@ impl DocumentsInner {
           .map(|im| JsxResolver::new(im, self.maybe_import_map.clone()))
       })
       .flatten();
-    if let Some(Ok(Some(imports))) =
-      maybe_config_file.map(|cf| cf.to_maybe_imports())
-    {
-      for (referrer, dependencies) in imports {
-        let dependencies =
-          dependencies.into_iter().map(|s| (s, None)).collect();
-        let module = SyntheticModule::new(
-          referrer.clone(),
-          dependencies,
-          self.get_maybe_resolver(),
-        );
-        self.imports.insert(referrer, module);
-      }
-    }
+    self.imports = Arc::new(
+      if let Some(Ok(Some(imports))) =
+        maybe_config_file.map(|cf| cf.to_maybe_imports())
+      {
+        imports
+          .into_iter()
+          .map(|(referrer, dependencies)| {
+            let dependencies =
+              dependencies.into_iter().map(|s| (s, None)).collect();
+            let module = SyntheticModule::new(
+              referrer.clone(),
+              dependencies,
+              self.get_maybe_resolver(),
+            );
+            (referrer, module)
+          })
+          .collect()
+      } else {
+        HashMap::new()
+      },
+    );
     self.dirty = true;
   }
 }
 
+fn calculate_fs_version(
```

Comment:
Both these functions were extracted out for reuse.

Reply:
nit: doc comments